script.js

```diff
@@ -1,4 +1,102 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+// Declare default output value as global variable
+var myOutputValue = 'hello world';
+// Declare variable tracking the number of attempts
+var attemptsCount = 0;
+// Declare variable tracking the number of wins by the user
+var winCount_user = 0;
+// Declare variable tracking the number of wins by the computer
+var winCount_com = 0;
+// Declare variable tracking the number of draws
+var drawCount = 0;
+// Declare 3 variables containing the string of 3 possible options
+var STONE = 'stone';
+var SCISSORS = 'scissors';
+var PAPER = 'paper';
+// Declare an array of the game options
+var gameOptions = [STONE, SCISSORS, PAPER];
+
+
+
+
+// Define a function that randomly generates either scissors, paper or stone
+var randomDraw = function () {
+  console.log('running randomDraw function');
+  
+  // Generate a random decimal between 0 to 3, inclusive of 0 and excluding 3
+  var randomDecimal = Math.random() * 3;
+  // Remove decimal to generate a random integer between 0 to 2, inclusive of 0 and 2
+  var randomInteger = Math.floor(randomDecimal);
+  // Assign string value of the array 'gameOptions' indexed by the random integer to the variable randomOutput
+  var randomOutput = gameOptions[randomInteger];
+
+  console.log('randomOutput');
+  console.log(randomOutput);
+  // Return the randomly chosen option, scissors paper or stone
+  return randomOutput;
+};
+
+
+
+// Define a function that checks if the user input is valid i.e. scissors, paper or stone
+var checkInput = function (str) {
+  // Convert input string characters to lowercase
+  var input = str.toLowerCase();
```

Comment:
We already convert the string to lowercase in line 55. is there a reason why we are doing it again ?

Reply:
Thanks for pointing this out. Here this conversion is a local variable within the checkInput function, so the conversion only occurs within this function when it is called to check the input, but I think it does not convert the 'input' in the main function in line 55 as the 'input' in line 55 is technically a different local variable, correct? Let me know if my reasoning makes sense or if I have some misunderstanding